### ISSUE TITLE
C# Vector3.cs Slerp() Method Normalize Error

### ISSUE THREAD

**Issue by Na-r** [godotengine/godot#94049]
### Tested versions

- 4.2.2.stable.mono
- 4.3.beta2.mono

### System information

Godot v4.2.2.stable.mono - NixOS #1-NixOS SMP PREEMPT_DYNAMIC Fri May 17 10:02:40 UTC 2024 - X11 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4090 - AMD Ryzen 9 5900X 12-Core Processor (24 Threads)

### Issue description

The Slerp function in Vector3.cs throws errors to seemingly normal input. Here's the error and an example of a call that caused it. This issue is notably absent from Vector2.cs.

```
Vector3 vecA = new Vector3(0.12f, 0.12f, 0.12f);
Vector3 vecB = new Vector3(0.512f, 0.512f, 0.512f);
vecA.Slerp(vecB, 0.5f);;
```
```
Vector3.cs:627 @ Godot.Vector3 Godot.Vector3.Rotated(Godot.Vector3, float): System.ArgumentException: Argument is not normalized. (Parameter 'axis')
  <C# Error>     System.ArgumentException
  <C# Source>    /root/godot/modules/mono/glue/GodotSharp/GodotSharp/Core/Vector3.cs:627 @ Godot.Vector3 Godot.Vector3.Rotated(Godot.Vector3, float)
  <Stack Trace>  Vector3.cs:627 @ Godot.Vector3 Godot.Vector3.Rotated(Godot.Vector3, float)
                 Vector3.cs:698 @ Godot.Vector3 Godot.Vector3.Slerp(Godot.Vector3, float)
                 Test.cs:9 @ void Test._Ready()
                 Node.cs:2401 @ bool Godot.Node.InvokeGodotClassMethod(Godot.NativeInterop.godot_string_name&, Godot.NativeInterop.NativeVariantPtrArgs, Godot.NativeInterop.godot_variant&)
                 CanvasItem.cs:1504 @ bool Godot.CanvasItem.InvokeGodotClassMethod(Godot.NativeInterop.godot_string_name&, Godot.NativeInterop.NativeVariantPtrArgs, Godot.NativeInterop.godot_variant&)
                 Node2D.cs:546 @ bool Godot.Node2D.InvokeGodotClassMethod(Godot.NativeInterop.godot_string_name&, Godot.NativeInterop.NativeVariantPtrArgs, Godot.NativeInterop.godot_variant&)
                 Test_ScriptMethods.generated.cs:38 @ bool Test.InvokeGodotClassMethod(Godot.NativeInterop.godot_string_name&, Godot.NativeInterop.NativeVariantPtrArgs, Godot.NativeInterop.godot_variant&)
                 CSharpInstanceBridge.cs:24 @ Godot.NativeInterop.godot_bool Godot.Bridge.CSharpInstanceBridge.Call(nint, Godot.NativeInterop.godot_string_name*, Godot.NativeInterop.godot_variant**, int, Godot.NativeInterop.godot_variant_call_error*, Godot.NativeInterop.godot_variant*)
```

In my testing, it seems like this error is only thrown when X=Y=Z for each Vector3. Changing any of the members at all prevents the error. This doesn't seem like an erroneous use case though.

The call to Vector3.Rotated is apparently always normalized: [Vector3.cs](https://github.com/godotengine/godot/blob/f3af22b10b1e64146d48be7726f6395c9a4185e9/modules/mono/glue/GodotSharp/GodotSharp/Core/Vector3.cs#L698)

### Steps to reproduce

Call Vector3.Slerp(), where all the members of each Vector3 are set to the same values.

### Minimal reproduction project (MRP)

[MRP.zip](https://github.com/user-attachments/files/16120636/MRP.zip)

**Comment by raulsntos** [comment]
As you noticed, the C# implementation of `Vector3.Rotated` checks that the `axis` parameter is normalized:

https://github.com/godotengine/godot/blob/a48b99af8d141590d326e93321adec53c5244826/modules/mono/glue/GodotSharp/GodotSharp/Core/Vector3.cs#L622-L631

It looks like the C++ implementation doesn't have this check:

https://github.com/godotengine/godot/blob/a48b99af8d141590d326e93321adec53c5244826/core/math/vector3.cpp#L38-L46

The C# implementation should match Core, so we should remove this check from C#. It's also worth checking that other C# methods don't have extra or missing checks. For example, `Vector3.Slide` is not checking that the `normal` parameter is normalized in C#, but the check exists in C++.

> The call to Vector3.Rotated is apparently always normalized

It seems the vector is `(0, 0, 0)` which can't be normalized.

**Comment by kleonc** [comment]
> It looks like the C++ implementation doesn't have this check:
> 
> https://github.com/godotengine/godot/blob/a48b99af8d141590d326e93321adec53c5244826/core/math/vector3.cpp#L38-L46

The `Basis(p_axis, p_angle)` part includes axis checking:
https://github.com/godotengine/godot/blob/8e36f98ea5b5ab4c1fb5249f94b61a7bbfb379a1/core/math/basis.h#L229
https://github.com/godotengine/godot/blob/8e36f98ea5b5ab4c1fb5249f94b61a7bbfb379a1/core/math/basis.cpp#L849-L853

However, the same constructor in C# doesn't have such checking even though it's said that `axis` must be normalized:
https://github.com/godotengine/godot/blob/8e36f98ea5b5ab4c1fb5249f94b61a7bbfb379a1/modules/mono/glue/GodotSharp/GodotSharp/Core/Basis.cs#L871-L902

So there are mismatches between Core and C# indeed.

**Comment by aaronfranke** [comment]
Apologies, I fixed this in C++ in #63912 but forgot to do the same fix in C#. Thanks @Z0rb14n for fixing it in #94766.
